src/app/pagination/pagination.component.html

```diff
@@ -0,0 +1,31 @@
+<div class="content-view-pf-pagination list-view-pf-pagination clearfix">
+  <div class="form-group">
+    <div class="padding-right-10">
+      <div class="btn-group dropdown" dropdown>
+        <button class="btn btn-default dropdown-toggle" dropdownToggle>
+        {{config.pageSize}}
+        <span class="caret"></span>  
+      </button>
+        <ul class="dropdown-menu" *dropdownMenu>
+          <li *ngFor="let increment of config.pageSizeIncrements" [ngClass]="{'selected': increment === config.pageSize}" class="display-length-increment">
+            <a role="menuitem" (click)="onPageSizeChange(increment)">{{increment}}</a></li>
+        </ul>
+      </div>
+    </div>
+    <span class="per-page-label">per page</span>
+  </div>
+  <div class="form-group">
+    <span><span class="pagination-pf-item-current">{{getStartIndex()}} - {{getEndIndex()}}</span> of <span class="pagination-pf-items-total">{{config.numTotalItems}}</span></span>
+    <ul class="pagination pagination-pf-back">
+      <li [ngClass]="{'disabled': config.pageNumber === 1}"><a title="First Page" (click)="gotoFirstPage()" class="goto-first-page"><span class="i fa fa-angle-double-left"></span></a></li>
+      <li [ngClass]="{'disabled': config.pageNumber === 1}"><a title="Previous Page" (click)="gotoPreviousPage()" class="goto-prev-page"><span class="i fa fa-angle-left"></span></a></li>
+    </ul>
+    <input type="text" #pageNoInput [ngModel]="config.pageNumber" (keyup.enter)="onPageNumberChange(pageNoInput.value)" class="pagination-pf-page"
```

Comment:
We could use (keyup.enter)=""; however, that will always generate an event regardless if the value changed -- whenever enter is pressed.

I was also testing the Angular Patternfly example and noticed that they use the blur event. I assume that was done for the case where the user changes the page number, does not press enter, and navigates away from the input field. If pagination isn't changed, the input field would be left with an incorrect value. 

We should ensure that test case is handled as well -- the change event will do that, but keyup.enter probably will not?

In either case, we will still want to use [(ngModel)] instead of passing the input value to onPageNumberChange. I tried this out with your code and it appears to work as expected.

The two way binding will also allow devs to set the current page, via the config.pageNumber, if desired.